ngraph_bridge/ngraph_pipelined_tensors.cc

```
@@ -91,30 +91,23 @@ bool IndexLibrary::is_free(size_t id) {
 
 PipelinedTensorsStore::PipelinedTensorsStore(PipelinedTensorMatrix in,
                                              PipelinedTensorMatrix out)
-    : m_in_tensors(in),
-      m_out_tensors(out),
-      m_num_inputs(in.size()),
-      m_num_outputs(out.size()) {
+    : m_in_tensors(in), m_out_tensors(out) {
+  auto m_depth_in = in.size();
+  auto m_depth_out = out.size();
+
+  // We assume that input and output depths are same
+  m_depth = std::min(m_depth_in, m_depth_out);
```

Lets throw an error here